ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java

```diff
@@ -1804,9 +1803,9 @@ private static String normalizeDateCol(Object colValue) throws SemanticException
       throw new SemanticException("Unexpected date type " + colValue.getClass());
     }
     try {
-      return MetaStoreUtils.PARTITION_DATE_FORMAT.get().format(
-          MetaStoreUtils.PARTITION_DATE_FORMAT.get().parse(value.toString()));
-    } catch (ParseException e) {
+      return MetaStoreUtils.convertDateToString(
+          MetaStoreUtils.convertStringToDate(value.toString()));
+    } catch (Exception e) {
       throw new SemanticException(e);
     }
   }
```

Comment:
Interestingly this seems to be dead code. We probably don't need to do something as part of this PR but I was wondering why it was left here. We can investigate later on after we are done with this.

Reply:
In case of `ParseException`, they were catching it and throwing `SemanticException`, now new code don't throw any exception in normal scenario so for the safer side catching `Exception` and throwing back `SemanticException`.

Reply:
Yeah, I could see only testcode is calling it. If required we can remove it(anyway as you mentioned its not related to the current PR).